tutorials/manage-instances-with-terraform-and-functions/index.mdx

```diff
@@ -0,0 +1,311 @@
+    - `provider.tf`: Terraform relies on plugins called “[providers](https://www.terraform.io/language/providers)” to interact with remote systems
+    - `backend.tf`: each Terraform configuration can specify a backend, which defines where the state file of the current infrastructure will be stored. Terraform uses this file to keep track of the managed resources. This state can be stored locally or remotely. Configuring a remote backend allows multiple people to work on the same infrastructure
+    - `variables.tf`: will contain the variable definitions for your project. Since all Terraform values must be defined, any variables that are not given a default value will become required arguments
+    - `terraform.tfvars`: allows you to set the actual value of the variables
+
+3. Create the following folders:
+    - `function`: to store your function code
+    - `files`: to temporary store your zip function code
+    Your folder structure should look like:
+    ```bash
+    Terraform
+    | -- files
+    | -- function
+    -- main.tf
+    -- backend.tf
+    -- provider.tf
+    -- terraform.tfvars
+    -- variables.tf
+    ```
+
+4. Edit the `backend.tf` file to enable remote configuration backup:
+    ```json
+        terraform {
+        backend "s3" {
+            bucket                      = "XXXXXXXXX"
+            key                         = "terraform.tfstate"
+            region                      = "fr-par"
+            endpoint                    = "https://s3.fr-par.scw.cloud"
+            skip_credentials_validation = true
+            skip_region_validation      = true
+        }
+        }
+        /*
+        For the credentials part:
+        ==> Create a ~/.aws/credentials:
+        [default]
+        aws_access_key_id=<SCW_ACCESS_KEY>
+        aws_secret_access_key=<SCW_SECRET_KEY>
+        region=fr-par
+        */
+    ```
+
+5. Edit the `provider.tf` file and add Scaleway as a provider:
+    ```json
+    terraform {
+      required_providers {
+        scaleway = {
+          source  = "scaleway/scaleway"
+          version = "2.2.7"
+        }
+      }
+      required_version = ">= 0.13"
+    }
+    ```
+
+  6. Specify the following variables in the `variables.tf` file
+    ```json
+    variable "zone" {
+      type = string
+    }
+
+    variable "region" {
+      type = string
+    }
+
+    variable "env" {
+      type = string
+    }
+
+    variable "project_id" {
+      type        = string
+      description = "Your project ID"
+    }
+
+    variable "auth_token" {
+        type = string
+        description = "Scaleway authtentication token used in the function"
+    }
+    ```
+
+6. Add the variable values to `terraform.tfvars` 
+    ```bash
+    zone                           = "fr-par-1"
+    region                         = "fr-par"
+    env                            = "dev"
+    project_id                     = "Your project ID"
+    auth_token                     = "Your Scaleway Auth Token"
```

Comment:
Isn't it a little bit confusing in terms of vocabulary about "Auth Token" vs  "Secret Key" etc... ?